File: api/client-samples/ruby/sample.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

##
# NameDivider API Ruby SDK サンプルコード

require_relative 'sdk_like'

##
# NameDivider API SDK の使用例
def main
  client = NameDividerClient.new('http://localhost:8000')
  test_names = %w[原敬 菅義偉 安倍晋三 中曽根康弘]

  begin
    # 専用メソッド divide_basic() を使用
    basic_results = client.divide_basic(test_names)
    puts 'BasicNameDivider Results:'
    basic_results.each_with_index do |result, i|
      puts "#{i + 1}. #{test_names[i]} -> #{result.family} #{result.given} (score: #{result.score.round(4)})"
    end

    # 専用メソッド divide_gbdt() を使用
    puts "\nGBDTNameDivider Results:"
    gbdt_results = client.divide_gbdt(test_names)
    gbdt_results.each_with_index do |result, i|
      puts "#{i + 1}. #{test_names[i]} -> #{result.family} #{result.given} (score: #{result.score.round(4)})"
    end

    # 汎用メソッド divide() とオプション指定
    puts "\n汎用メソッド使用例:"
    basic_option_results = client.divide(test_names, 'basic')
    gbdt_option_results = client.divide(test_names, 'gbdt')
    puts "Basic結果例: #{basic_option_results[0].family} #{basic_option_results[0].given}"
    puts "GBDT結果例: #{gbdt_option_results[0].family} #{gbdt_option_results[0].given}"

    # 単体の名前を処理
    single_result = client.divide_basic(['小泉純一郎'])
    puts "\n単体処理: 小泉純一郎 -> #{single_result[0].family} #{single_result[0].given}"

    # Rubyらしい関数型プログラミング例
    puts "\nRubyらしい処理例:"
    author_names = %w[田中角栄 吉田茂 池田勇人]
    author_results = client.divide_gbdt(author_names)
    
    # 高スコアの結果のみをフィルタリング
    high_confidence = author_results
                        .select { |result| result.score > 0.4 }
                        .map { |result| "#{result.family} #{result.given}" }
    
    puts "高スコア結果: #{high_confidence.join(', ')}" unless high_confidence.empty?

    puts "\nサンプル実行完了!"

  rescue StandardError => e
    puts "エラーが発生しました: #{e.message}"
    puts 'NameDivider APIサーバーが起動していることを確認してください。'
    puts '起動コマンド: docker run -d --rm -p 8000:8000 namedivider-api:v0.3.0-simple'
  end
end

# スクリプトが直接実行された場合のみmainを呼び出し
main if __FILE__ == $PROGRAM_NAME

File: api/client-samples/go/sample.go
/*
NameDivider API Go SDK サンプルコード
*/

package main

import (
	"fmt"
	"log"
)

/*
NameDivider API SDK の使用例
*/
func main() {
	client := NewNameDividerClient("http://localhost:8000")
	testNames := []string{"原敬", "菅義偉", "安倍晋三", "中曽根康弘"}

	// 専用メソッド DivideBasic() を使用
	basicResults, err := client.DivideBasic(testNames)
	if err != nil {
		log.Printf("BasicNameDivider エラー: %v", err)
		return
	}

	fmt.Println("BasicNameDivider Results:")
	for i, result := range basicResults {
		fmt.Printf("%d. %s -> %s %s (score: %.4f)\n", i+1, testNames[i], result.Family, result.Given, result.Score)
	}

	// 専用メソッド DivideGBDT() を使用
	fmt.Println("\nGBDTNameDivider Results:")
	gbdtResults, err := client.DivideGBDT(testNames)
	if err != nil {
		log.Printf("GBDTNameDivider エラー: %v", err)
		return
	}

	for i, result := range gbdtResults {
		fmt.Printf("%d. %s -> %s %s (score: %.4f)\n", i+1, testNames[i], result.Family, result.Given, result.Score)
	}

	// 汎用メソッド Divide() とオプション指定
	fmt.Println("\n汎用メソッド使用例:")
	basicOptionResults, err := client.Divide(testNames, "basic")
	if err != nil {
		log.Printf("基本モード エラー: %v", err)
		return
	}

	gbdtOptionResults, err := client.Divide(testNames, "gbdt")
	if err != nil {
		log.Printf("GBDTモード エラー: %v", err)
		return
	}

	fmt.Printf("Basic結果例: %s %s\n", basicOptionResults[0].Family, basicOptionResults[0].Given)
	fmt.Printf("GBDT結果例: %s %s\n", gbdtOptionResults[0].Family, gbdtOptionResults[0].Given)

	// 単体の名前を処理
	singleResult, err := client.DivideBasic([]string{"小泉純一郎"})
	if err != nil {
		log.Printf("単体処理 エラー: %v", err)
		return
	}

	fmt.Printf("\n単体処理: 小泉純一郎 -> %s %s\n", singleResult[0].Family, singleResult[0].Given)

	fmt.Println("\nサンプル実行完了!")
}